### DISCUSSION TITLE
a question in Tutorial 09 - Privilege Level

### DISCUSSION THREAD

**Discussion by sunshine-lcc** [rust-embedded/rust-raspberrypi-OS-tutorials#146] [Tutorial Questions]
In the article in Tutorial 09 - Privilege Level, I realize how to switch to EL1, but I have a question that if I want to pass some parameter to EL1 function, like `kernel_init(arg1, arg2)`, what can I do for this? Could you please help me?

**Comment by andre-richter** [comment]
You‘ll need to put these arguments in the CPU registers before the `eret`.

arg1 == x0, arg2 == x1, etc.
